pkg/controller/operators/labeller/filters.go

```diff
@@ -0,0 +1,106 @@
+package labeller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/registry/reconciler"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/metadata"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/controller/operators/internal/alongside"
+)
+
+func Filter(gvr schema.GroupVersionResource) func(metav1.Object) bool {
+	if f, ok := filters[gvr]; ok {
+		return f
+	}
+	return func(object metav1.Object) bool {
+		return false
+	}
+}
+
+var filters = map[schema.GroupVersionResource]func(metav1.Object) bool{
+	corev1.SchemeGroupVersion.WithResource("services"): HasOLMOwnerRef,
+	corev1.SchemeGroupVersion.WithResource("pods"): func(object metav1.Object) bool {
+		if labels := object.GetLabels(); labels != nil {
+			if _, ok := labels[reconciler.CatalogSourceLabelKey]; ok {
+				return true
+			}
+		}
+		return false
+	},
+	corev1.SchemeGroupVersion.WithResource("serviceaccounts"): func(object metav1.Object) bool {
+		return HasOLMOwnerRef(object) || HasOLMLabel(object)
+	},
+	batchv1.SchemeGroupVersion.WithResource("jobs"): func(object metav1.Object) bool {
+		job, ok := object.(*batchv1.Job)
+		if !ok {
+			return false
+		}
+		for _, container := range job.Spec.Template.Spec.Containers {
+			if strings.Join(container.Command[0:3], " ") == "opm alpha bundle extract" {
+				return true
+			}
+		}
+		return false
+	},
+	appsv1.SchemeGroupVersion.WithResource("deployments"):          HasOLMOwnerRef,
+	rbacv1.SchemeGroupVersion.WithResource("roles"):                HasOLMOwnerRef,
+	rbacv1.SchemeGroupVersion.WithResource("rolebindings"):         HasOLMOwnerRef,
+	rbacv1.SchemeGroupVersion.WithResource("clusterroles"):         HasOLMOwnerRef,
+	rbacv1.SchemeGroupVersion.WithResource("clusterrolebindingss"): HasOLMOwnerRef,
```

Comment:
If I'm getting this right, we're applying filters to identify the resources OLM manages. Will all the current installations' managed resources have the necessary labels/annotations? I guess the right question is, what's the risk of this upgrading and losing track of previously managed resources? My thought it that is should low, but I don't know enough about the guts to have a high level of certainty there isn't something I'm missing =S

Reply:
It's a good question. I hope someone with more OLMv0 knowledge can chime in here - I've been doing ad-hoc testing on existing clusters with v0 and in my tests these filters are sufficient to find all existing data OLM created in the past, but I don't know if OLMv0 had every code path exercised there.

Reply:
@awgreene @joelanford @kevinrizza PTAL